### PR TITLE
reworks oxygen pathology symptoms

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
This changes the two oxygen related pathology symptoms, because they were always kinda super useless and did practically nothing.

Oxygen Conversion now basically gives you a buff that converts all the OXY damage you take into a smaller amount of BRUTE damage, it also gives a tiny speed boost during this, so you can walk away from what is probably a breach a little quicker.
The efficiency of the conversion increases as the pathogens stage increases.

Oxygen Production is now Oxygen Storage. It stores up to 100 oxygen while you are in an area where you can breathe. When you enter an area where you cannot, you receive a buff that has however many oxygen units were stored. At max storage, this will prevent up to 100 OXY damage until you run out. While the buff is active, you also have a small speed buff so you can vacate the area more quickly.
The speed of how quickly oxygen is stored during the downtime scales based on the pathogens stage.

Both Symptoms have been moved to the Tier 4 category rather than the Tier 5 category, seeing as they aren't really *that* good and that there aren't too many beneficial Tier 4 symptoms.

I made icons for the buffs you get too, beware the coder art:
image
image

## Why's this needed? 
- old symptoms were pretty garbage, especially for tier 5
- new should be more helpful, more interesting, and more noticeable for players
- adds more beneficial stuff in Tier 4

### CODE REVIEW

File: code/modules/medical/pathology/pathogen_benevolent.dm [hunk 1]
```diff
@@ -179,30 +179,37 @@ datum/pathogeneffects/benevolent/oxygenconversion
...
-		if (M.get_oxygen_deprivation())
-			if (origin.stage != 0)
-				M.take_oxygen_deprivation(0 - (origin.stage / 2))
-
-datum/pathogeneffects/benevolent/oxygenproduction
-	name = "Oxygen Production"
-	desc = "The pathogen produces oxygen."
-	rarity = RARITY_VERY_RARE
+		var/mob/living/carbon/C = M
+		var/oxy_damage = max(20, C.get_oxygen_deprivation())
```
Comment (Tarmunora) [inline code/modules/medical/pathology/pathogen_benevolent.dm:183]: I don't think you need this var

File: code/modules/medical/pathology/pathogen_benevolent.dm [hunk 2]
```diff
@@ -179,30 +179,37 @@ datum/pathogeneffects/benevolent/oxygenconversion
...
+		if (C.oxyloss || oxy_damage > 20)
+			var/datum/statusEffect/patho_oxy_speed/bad/POS = C.setStatus("patho_oxy_speed_bad", duration = INFINITE_STATUS)
+			POS.efficiency = origin.stage/2.5 // more efficient conversion at higher stages
```
Comment (Tarmunora) [inline code/modules/medical/pathology/pathogen_benevolent.dm:186]: could set `efficiency` with the `optional` arg on on setStatus

File: code/modules/medical/pathology/pathogen_benevolent.dm [hunk 3]
```diff
@@ -179,30 +179,37 @@ datum/pathogeneffects/benevolent/oxygenconversion
...
 			return
-		if (M:losebreath > 0)
-			M:losebreath = 0
-		if (M.get_oxygen_deprivation())
-			M.take_oxygen_deprivation(0 - origin.stage)
+		if(!origin.symptom_data["oxygen_storage"]) // if not yet set, initialize
+			origin.symptom_data["oxygen_storage"] = 0
+
+		var/mob/living/carbon/C = M
+		var/oxy_damage = max(20, C.get_oxygen_deprivation())
```
Comment (Tarmunora) [inline code/modules/medical/pathology/pathogen_benevolent.dm:203]: var isn't needed

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/medical/pathology/pathogen_benevolent.dm (hunk 2 above)
Comment (zjdtmkhzt) [reply code/modules/medical/pathology/pathogen_benevolent.dm]: Thanks, I must have missed that the setStatus had an optional argument!

**Comment by Tarmunora** [comment]
why not be checking `C.get_oxygen_deprivation() > 20`  in the conditionals, to match previous behavior?

**Comment by zjdtmkhzt** [comment]
Because it's not actually necessary, it was left over from a previous iteration.
For some reason people just randomly have 20 oxygen damage sometimes, which triggered the pathogen randomly.
Oxyloss doesn't have this problem, so it seemed to work fine without.

**Comment by Tarmunora** [comment]
```dm
/mob/living/carbon/get_oxygen_deprivation()
	return src.oxyloss
```
🤷

**Comment by zjdtmkhzt** [comment]
Yeah, I guess it does work with get_oxygen_deprivation!
I probably mixed up min and max at some point during early testing, for some reason I always do that^^

Thanks for all your help, yass!

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**